mantidproject/mantid
# Invalid detector list message popping up continually

~~This should be tested and merged after #26299~~

**Description of work.**

Currently when an invalid detector list is loaded the user is repeatedly warned as the validator is triggered multiple times by a single event. This updates the table items so that the validator is added later meaning the user is only warned about invalid values once.

**To test:**
  * In Muon Analysis load some data, any of the EMU files from the unit test directory should do.
  * Go to the grouping tab and attempt to modify the groups and pairs to have invalid (generally negative or too large) values check that a single box pops up warning you of the error before reverting the change.
  * Check that you cannot get some invalid groups into the table either manually or by attempting to load in invalid grouping tables. You can create an invalid table by saving out the current one and then manually editing it before loading it back in.



*There is no associated issue.*

*This does not require release notes* because this fixes issues introduced in this release.


---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
